packages/xstate-react/src/useActor.ts

```diff
@@ -1,12 +1,16 @@
 import { useState, useRef } from 'react';
 import useIsomorphicLayoutEffect from 'use-isomorphic-layout-effect';
 import { Sender, ActorRefLike } from './types';
-import { EventObject, Actor } from 'xstate';
+import { EventObject } from 'xstate';
 import useConstant from './useConstant';
 
-export function useActor<TEvent extends EventObject, TEmitted = any>(
-  actorRef: ActorRefLike<TEvent, TEmitted> | Actor,
-  getSnapshot: (actor: typeof actorRef) => TEmitted = (a) =>
+export function useActor<
+  TEvent extends EventObject,
+  TEmitted = unknown,
+  TActor extends ActorRefLike<TEvent, TEmitted> = ActorRefLike<TEvent, TEmitted>
+>(
+  actorRef: TActor & ActorRefLike<TEvent, TEmitted>,
+  getSnapshot: (actor: TActor) => TEmitted = (a) =>
```

Comment:
Could you describe what you have been trying to do here? It's a little bit mind-bending and I'm lost here. I don't understand why `TActor` has been introduced and why `ActorRefLike<...>` is not just used directly. When one would like to provide a subtype of that `ActorRefLike` type? I also don't understand what `TActor & ActorRefLike<TEvent, TEmitted>` is supposed to do 🤔

Reply:
Try removing `ActorRefLike<...>`. The `TEmitted` generic param becomes `unknown` which makes TS fail to properly infer the `actorRef`.

`ActorRefLike<...>` cannot be used directly because the `actor` argument in `getSnapshot` must match the inferred `actorRef` argument.